src/farcaster/getFarcasterUserAddress.ts

```diff
@@ -0,0 +1,48 @@
+import { getCustodyAddressForFidNeynar } from '../utils/neynar/user/getCustodyAddressForFidNeynar';
+import { getVerifiedAddressesForFidNeynar } from '../utils/neynar/user/getVerifiedAddressesForFidNeynar';
+import { GetFarcasterUserAddressResponse } from './types';
+
+type GetFarcasterUserAddressOptions =
+  | {
+      neynarApiKey?: string; // default to onchain-kit's default key
+      hasCustodyAddresses?: boolean; // default to true
+      hasVerifiedAddresses?: boolean; // default to true
+    }
+  | undefined;
+
+/**
+ * Get the user address for a given fid
+ * @param fid  The farcaster id
+ * @param GetFarcasterUserAddressOptions  The options to specify the type of addresses to get and the neynar api key
+ * @returns the custory address and/or verified addresses. If there is an error, it returns null
+ */
+async function getFarcasterUserAddress(
+  fid: number,
+  options?: GetFarcasterUserAddressOptions,
+): Promise<GetFarcasterUserAddressResponse | null> {
+  try {
+    const hasCustodyAddresses = options?.hasCustodyAddresses ?? true;
```

Comment:
`hasCustodyAddresses` @Sneh1999 did you mentioned there can be only one address for custody?

Reply:
So why not `hasCustodyAddress` as singular?